core/src/main/java/io/smallrye/openapi/api/SmallRyeOpenAPI.java

```diff
@@ -564,10 +565,11 @@ protected <V, A extends V, O extends V, AB, OB> void buildStaticModel(BuildConte
                             return ctx.appClassLoader.getResource(loadPath.toString());
                         });
 
-                for (URL staticFile : OpenApiProcessor.locateStaticFiles(loadFn)) {
+                for (OpenApiStaticFile staticFile : OpenApiProcessor.loadOpenApiStaticFiles(loadFn)) {
```

Comment:
Using `OpenApiStaticFile` will be problematic because it takes the open `InputStream` (which may cause file locking on Windows). Maybe we can keep using `locateStaticFiles`, but determine the `Format` based on whether it ends with `.json`.